File: developer/Samples/NET 4.6/WF_WCF_Samples/WF/Basic/Rules/Policy/SimplePolicy/CS/SimplePolicy/SimplePolicyWorkflow.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System;
using System.Workflow.Activities;

namespace Microsoft.Samples.Workflow.SimplePolicy
{
    public sealed partial class SimplePolicyWorkflow : SequentialWorkflowActivity
    {
        // Set orderValue and customerType to change the discount calculated by the policy
        private double orderValue = 600.00;
        private CustomerType customerType = CustomerType.Residential;
        private double discount = 0;

        public SimplePolicyWorkflow()
        {
            InitializeComponent();
        }
        
        private void WorkflowCompleted(object sender, EventArgs e)
        {
            Console.WriteLine("Order value = {0:c}", orderValue);
            Console.WriteLine("Customer type = {0}", customerType);
            Console.WriteLine("Calculated discount = {0}%", discount);
        }

        public enum CustomerType
        {
            Empty = 0,
            Residential = 1,
            Business = 2
        }
    }
}

File: developer/Samples/NET 4.6/WF_WCF_Samples/WF/Basic/Rules/Policy/SimplePolicy/VB/SimplePolicy/SimplePolicyWorkflow.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------

Imports System
Imports System.CodeDom
Imports System.ComponentModel
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

Namespace Microsoft.Samples.Workflow.SimplePolicy
    Public Class SimplePolicyWorkflow
        Inherits SequentialWorkflowActivity

        ' Set orderValue and customerTypeValue to change the discount calculated by the policy
        Private orderValue As Double = 600.0
        Private customerTypeValue As CustomerType = CustomerType.Residential
        Private discount As Double = 0

        Private Sub OnWorkflowCompleted(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Order value = {0:c}", orderValue)
            Console.WriteLine("Customer type = {0}", customerTypeValue)
            Console.WriteLine("Calculated discount = {0}%", discount)
        End Sub

        Public Enum CustomerType
            Empty = 0
            Residential = 1
            Business = 2
        End Enum
    End Class
End Namespace
